### PR TITLE
New versions: py-sphinxcontrib-bibtex, py-sphinx-rtd-theme, py-pybtex-docutils, py-pybtex

### PR DESCRIPTION
**Pull request by prckent**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-pybtex-docutils/package.py [hunk 1]
```diff
@@ -11,6 +11,7 @@ class PyPybtexDocutils(PythonPackage):
 
     pypi = "pybtex-docutils/pybtex-docutils-0.2.1.tar.gz"
 
+    version('1.0.0', sha256='cead6554b4af99c287dd29f38b1fa152c9542f56a51cb6cbc3997c95b2725b2e')
     version('0.2.2', sha256='ea90935da188a0f4de2fe6b32930e185c33a0e306154322ccc12e519ebb5fa7d')
     version('0.2.1', sha256='e4b075641c1d68a3e98a6d73ad3d029293fcf9e0773512315ef9c8482f251337')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pybtex-docutils/package.py:17]: Latest version requires Python 3.6+, and six is no longer required

File: var/spack/repos/builtin/packages/py-pybtex/package.py [hunk 2]
```diff
@@ -12,9 +12,10 @@ class PyPybtex(PythonPackage):
        Python."""
 
     homepage = "https://pybtex.org"
-    pypi = "Pybtex/pybtex-0.21.tar.gz"
+    pypi = "pybtex/pybtex-0.24.0.tar.gz"
 
-    version('0.21', sha256='af8a6c7c74954ad305553b118d2757f68bc77c5dd5d5de2cc1fd16db90046000')
+    version('0.24.0', sha256='818eae35b61733e5c007c3fcd2cfb75ed1bc8b4173c1f70b56cc4c0802d34755')
+    version('0.21.0', sha256='af8a6c7c74954ad305553b118d2757f68bc77c5dd5d5de2cc1fd16db90046000')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pybtex/package.py]:
```suggestion
    version('0.21', sha256='af8a6c7c74954ad305553b118d2757f68bc77c5dd5d5de2cc1fd16db90046000')
```

File: var/spack/repos/builtin/packages/py-pybtex/package.py [hunk 3]
```diff
@@ -12,9 +12,10 @@ class PyPybtex(PythonPackage):
...
 
     depends_on('py-setuptools', type='build')
     depends_on('py-latexcodec@1.0.4:', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pybtex/package.py:22]: Six is now required, and the Python dep is now `@2.7:2.8,3.3:`

File: var/spack/repos/builtin/packages/py-sphinx-rtd-theme/package.py [hunk 4]
```diff
@@ -12,6 +12,7 @@ class PySphinxRtdTheme(PythonPackage):
     homepage = "https://github.com/readthedocs/sphinx_rtd_theme"
     pypi = "sphinx-rtd-theme/sphinx_rtd_theme-0.5.1.tar.gz"
 
+    version('0.5.2', sha256='32bd3b5d13dc8186d7a42fc816a23d32e83a4827d7d9882948e7b837c232da5a')
     version('0.5.1', sha256='eda689eda0c7301a80cf122dad28b1861e5605cbf455558f3775e1e8200e83a5')
     version('0.5.0', sha256='22c795ba2832a169ca301cd0a083f7a434e09c538c70beb42782c073651b707d')
     version('0.4.3', sha256='728607e34d60456d736cc7991fd236afb828b21b82f956c5ea75f94c8414040a')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-sphinx-rtd-theme/package.py:18]: py-docutils@:0.16 is now required

File: var/spack/repos/builtin/packages/py-sphinxcontrib-bibtex/package.py [hunk 5]
```diff
@@ -11,6 +11,9 @@ class PySphinxcontribBibtex(PythonPackage):
 
     pypi = "sphinxcontrib-bibtex/sphinxcontrib-bibtex-0.3.5.tar.gz"
 
+    version('2.2.1', sha256='00d474092e04b1d941e645cf6c027632a975cd0b9337cf47d379f63a5928f334')
+    version('2.2.0', sha256='7500843e154d76983c23bca5ca7380965e0725c46b8f484c1322d0b58a6ce3b2')
+    version('2.1.4', sha256='f53ec0cd534d2c8f0a51b4b3473ced46e9cb0dd99a7c5019249fe0ef9cbef18e')
     version('2.0.0', sha256='98e18eb0b088d3f556199f3fbb91d3d48ebb7596fe86b6c37cc4c4dc5419b7a1')
     version('1.0.0', sha256='629612b001f86784669d65e662377a482052decfd9a0a17c46860878eef7b9e0')
     version('0.3.5', sha256='c93e2b4a0d14f0ab726f95f0a33c1675965e9df3ed04839635577b8f978206cd')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-sphinxcontrib-bibtex/package.py:19]: Need to update dependencies:
```python
Sphinx>=2.1
docutils>=0.8
pybtex>=0.20
pybtex-docutils>=1.0.0
dataclasses; python_version < '3.7'
```

**Review by adamjstewart — CHANGES_REQUESTED** [review]

**Comment by prckent** [comment]
Updated. Check dataclasses line for py-sphinxcontrib-bibtex

File: var/spack/repos/builtin/packages/py-sphinxcontrib-bibtex/package.py [hunk 6]
```diff
@@ -24,8 +27,12 @@ class PySphinxcontribBibtex(PythonPackage):
     depends_on('py-pybtex@0.20:', type=('build', 'run'), when='@2.0.0:')
     depends_on('py-pybtex-docutils@0.2.0:', type=('build', 'run'), when='@:1.9.9')
     depends_on('py-pybtex-docutils@0.2.2:', type=('build', 'run'), when='@2.0.0:')
+    depends_on('py-pybtex-docutils@1.0.0:', type=('build', 'run'), when='@2.2.0:')
     depends_on('py-six@1.4.1:', type=('build', 'run'), when='@0.3.5')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-sphinxcontrib-bibtex/package.py:31]: I don't think latexcodec or six or oset are required in the latest version. Was docutils required in older versions?

Comment (prckent) [reply var/spack/repos/builtin/packages/py-sphinxcontrib-bibtex/package.py:31]: Updated. A couple less build dependencies for the latest version as a result.

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**